src/main/java/classify/data/DataReader.java

```diff
@@ -76,6 +77,23 @@ public static void restoreStudentList(String studentFileInput, ArrayList <Studen
 
             //Set remarks
             student.getAttributes().setRemarks(inputArr[4].trim());
+
+            //Get all Subject Info
+            String [] allSubjects = (inputArr[5].trim()).split("#--#");
```

Comment:
Good choice of regex to separate subjects when there student is enrolled into multiple classes.